### PR TITLE
Node delete added in osdconfig

### PR DESCRIPTION
**Pull request by sdeoras**
**What this PR does / why we need it**:
Added node delete method in osdconfig interface. It is required for cleanup of data that in etcd

### CODE REVIEW

File: api/client/cluster/client.go [hunk 1]
```diff
@@ -324,3 +324,11 @@ func (c *clusterClient) SetNodeConf(config *osdconfig.NodeConfig) error {
 	}
 	return nil
 }
+
+func (c *clusterClient) DelNodeConf(nodeID string) error {
```
Comment (lpabon) [inline api/client/cluster/client.go]: If delete is the opposite of `SetNode`, then it should be `Unset`. Or change `SetNodeConf` to `CreateNodeConf` and then the opposite would be `DeleteNodeConf`

File: osdconfig/api_test.go [hunk 2]
```diff
@@ -75,6 +75,17 @@ func TestSetGetNode(t *testing.T) {
 	if !reflect.DeepEqual(expectedConf, receivedConf) {
 		t.Fatal("expected and received values are not deep equal")
 	}
+
+	// now delete the node
+	if err := manager.DelNodeConf(expectedConf.NodeId); err != nil {
+		t.Fatal("error in deleting node config")
```
Comment (lpabon) [inline osdconfig/api_test.go]: I see this code uses `t.Fatal()` a lot, but instead start using `testify.assert(...)`

**Comment by sdeoras** [comment]
@lpabon i am merging this PR since need it to close another one elsewhere. I already made the changes you requested.

File: osdconfig/api.go [hunk 3]
```diff
@@ -74,6 +74,20 @@ func (manager *configManager) SetNodeConf(config *NodeConfig) error {
 	return err
 }
 
+// UnsetNodeConf deletes node config data in kvdb
+func (manager *configManager) UnsetNodeConf(nodeID string) error {
+	manager.Lock()
```
Comment (sangleganesh) [inline osdconfig/api.go]: do we need the lock here ?

**Pull request merged**